### DISCUSSION TITLE
How to remove a reaction

### DISCUSSION THREAD

**Discussion by open-sudo** [Auties00/WhatsappWeb4j#300] [Q&A]
I see there is a method Whatsapp.removeReaction(). But I am not able to find out how to create the MessageInfo that it requires.  I also tried to modify a reaction that the user previously sent by removing its content and reposting it back into the chat. Like this:

```
  ((ReactionMessage)info.message().content()).content(null);
  whatsapp.removeReaction(reacted);
```
This did not help either. Thoughts?

**Comment by Auties00** [comment]
> hod Whatsapp.removeReaction(). But I am not able to find out how to create the MessageInfo that it requires

You can't create that message info, you need to query it from the store

**Comment by open-sudo** [comment]
So, in my case, the user sent a reaction ReactionMessage, say because they liked a message in the chat. How do I then remove that reaction? Should I take the reaction that the user sent and change the content and send it back? Is it how I am supposed to use removeAction()?

I am struggling to see how I would get the proper message info into the store in the first place.

**Reply by Auties00** [reply]
> So, in my case, the user sent a reaction ReactionMessage, say because they liked a message in the chat. How do I then remove that reaction? Should I take the reaction that the user sent and change the content and send it back? Is it how I am supposed to use removeAction()?
> 
> I am struggling to see how I would get the proper message info into the store in the first place.

onNewMessage listener
get the message
check if it's a reaction
if it is get the key to the message it references
store.findMessageByKey(key);

**Reply by open-sudo** [reply]
Either I am completely missing it or sendReaction() has an issue. I tried a simple thing:

1) Client sends a message
2) Whatsapp4j sends a reaction to that exact the same message.

That doesn't work. No reaction is sent shown on the client. I don't know if this is a client version issue. I run 2.23.9.75 on device and 2.2320.10 on computer. None is showing the reaction. And I see that removeReaction is internally calling sendReaction.
